kuma/wiki/templatetags/ssr.py

```diff
@@ -1,15 +1,31 @@
 from __future__ import print_function
 
 import json
+import os
 
 import requests
 import requests.exceptions
 from django.conf import settings
+from django.utils import lru_cache
 from django_jinja import library
 
 
+@lru_cache.lru_cache()
+def get_localization_data(locale):
+    """
+    Read the frontend string catalog for the specified locale, parse
+    it as JSON, and return the resulting dict. The returned values
+    are cached so that we don't have to read files all the time.
+    """
+    path = os.path.join(settings.BASE_DIR,
+                        'static', 'jsi18n',
+                        locale, 'react.json')
+    with open(path, 'r') as f:
+        return json.load(f)
+
+
 @library.global_function
-def render_react_app(document_data, request_data, ssr=True):
+def render_react_app(locale, document_data, request_data, ssr=True):
```

Comment:
I would vote for dumping `request_data` from the signature of this function, and then:
- change the creation of the `data` dictionary below to:
```
    data = {
        'localizationData': get_localization_data(locale),
        'documentData': document_data,
    }
```
- remove `request_data` altogether from the `react_document` view (and everywhere else it's used)
- only use `getLocale()` within the React components (dump all use of `data.requestData.locale`)

Reply:
I thought about doing that, but worry that I'll end up needing other data (I don't know what) from the request and won't have anywhere to put it.

Also, in order to make this work, I'd have to do:

```
localizationData['locale'] = locale
```